File: http-process/src/main/java/com/github/dragonhht/http/response/ResponseOutputStream.java
package com.github.dragonhht.http.response;

import javax.servlet.ServletOutputStream;
import javax.servlet.WriteListener;
import java.io.IOException;
import java.io.OutputStream;

/**
 * 描述.
 *
 * @author: huang
 * @Date: 18-11-11
 */
public class ResponseOutputStream extends ServletOutputStream {

    private OutputStream out;

    public ResponseOutputStream(OutputStream out) {
        this.out = out;
    }

    @Override
    public boolean isReady() {
        // TODO 待实现
        return false;
    }

    @Override
    public void setWriteListener(WriteListener writeListener) {
        // TODO 待实现
    }

    @Override
    public void write(int b) throws IOException {
        // TODO 待实现
    }

    @Override
    public void write(byte[] b) throws IOException {
        this.out.write(b);
    }

    @Override
    public void write(byte[] b, int off, int len) throws IOException {
        this.out.write(b, off, len);
    }

    @Override
    public void flush() throws IOException {
        this.out.flush();
    }

    @Override
    public void close() throws IOException {
        this.out.close();
    }
}

File: netty-server/src/main/java/com/github/dragonhht/http/servlet/ResponseOutputStream.kt
package com.github.dragonhht.http.servlet

import java.io.IOException
import java.io.OutputStream
import javax.servlet.ServletOutputStream
import javax.servlet.WriteListener

/**
 * .
 *
 * @author: huang
 * @Date: 2019-4-17
 */
class ResponseOutputStream(private val out: OutputStream) : ServletOutputStream() {

    override fun isReady(): Boolean {
        // TODO 待实现
        return false
    }

    override fun setWriteListener(writeListener: WriteListener) {
        // TODO 待实现
    }

    @Throws(IOException::class)
    override fun write(b: Int) {
        // TODO 待实现
    }

    @Throws(IOException::class)
    override fun write(b: ByteArray) {
        this.out.write(b)
    }

    @Throws(IOException::class)
    override fun write(b: ByteArray, off: Int, len: Int) {
        this.out.write(b, off, len)
    }

    @Throws(IOException::class)
    override fun flush() {
        this.out.flush()
    }

    @Throws(IOException::class)
    override fun close() {
        this.out.close()
    }
}